lib/compile.js

```diff
@@ -61,6 +60,7 @@ function Compile(structure, opts) {
                 Object.keys(c.macros).forEach(function(k) {
                     result.macros[k] = c.macros[k];
                 });
+                require('./lint.js')(result, {out: process.stderr});
```

Comment:
Even though this fixes the issue, I think it still breaks `--quiet` option. Does it need to lint inside compile at all?

Reply:
Hmm. It looks like `compile.js` only does this in order to lint `@include`-d files.

Maybe we should rewrite `lint.js` to recursively process included files by itself; I don't think we should be requiring `lint` from inside `compile`.

Reply:
...`lint` appears to only run on the generated grammar. Now I'm _very_ confused!

Reply:
What do you mean by that last statement? As it stands lint is run in `lib/compile.js` every time a new file is included and then [in nearleyc](https://github.com/Hardmath123/nearley/blob/aa77dee0d1fc728a5ffad046d4baed9eab49eaed/bin/nearleyc.js#L61).

Reply:
Do you mean it _should_ only be run on generated grammars but isn't or it should _not_ be run on generated grammars but is?